### PR TITLE
[xbmc][fixes] Get rid of a bunch of warnings about lossy conversions.

### PR DESCRIPTION
**Pull request by Paxxi**
Added 32-bit versions of asInteger to CVariant to cut down on the amount of casts needed.

Reminder that size_t is a variable size type, it's 64-bit on windows x64 so any return value from .size() will be an uint64_t on windows x64.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/Scraper.cpp [hunk 1]
```diff
@@ -655,8 +655,8 @@ PythonFind(const std::string& ID,
 
   if (XFILE::CDirectory::GetDirectory(str.str(), items))
   {
-    for (int i = 0; i < items.Size(); ++i)
-      result.emplace_back(std::move(FromFileItem<T>(*items[i])));
+      for (auto it : items)
+        result.emplace_back(std::move(FromFileItem<T>(*it)));
```
Comment (Rechi) [inline xbmc/addons/Scraper.cpp]: indentation

File: xbmc/addons/Scraper.cpp [hunk 2]
```diff
@@ -346,7 +346,7 @@ std::string CScraper::InternalRun(const std::string& function,
   // put the 'extra' parameterts into the parser parameter list too
   if (extras)
   {
-    for (unsigned int j=0;j<extras->size();++j)
+    for (size_t j=0;j<extras->size();++j)
```
Comment (notspiff) [inline xbmc/addons/Scraper.cpp]: Add spacing while at it.

File: xbmc/addons/Scraper.cpp [hunk 3]
```diff
@@ -447,7 +447,7 @@ CScraperUrl CScraper::NfoUrl(const std::string &sNfoContent)
 
   // parse returned XML: either <error> element on error, blank on failure,
   // or <url>...</url> or <url>...</url><id>...</id> on success
-  for (unsigned int i=0; i < vcsOut.size(); ++i)
+  for (size_t i=0; i < vcsOut.size(); ++i)
```
Comment (notspiff) [inline xbmc/addons/Scraper.cpp]: Dont see entire block but foreach?

File: xbmc/addons/Scraper.cpp [hunk 4]
```diff
@@ -523,7 +523,7 @@ CScraperUrl CScraper::ResolveIDToUrl(const std::string& externalID)
...
```
Comment (notspiff) [inline xbmc/addons/Scraper.cpp]: foreach?

**Review by notspiff — COMMENTED** [review]
Some small suggestions and a question; why 32 bit isnt it just the signs that are problematic?

File: xbmc/addons/Scraper.cpp (hunk 3 above)
Comment (Paxxi) [reply xbmc/addons/Scraper.cpp]: I checked but i is used for formatting in all the loops I changed

**Comment by Paxxi** [comment]
`CVariant::asInteger()` returns an int64_t so everywhere it's assigned to int VS will emit a possible truncation warning. First too was to add casts but I think this will be an issue in more places so added the new method to CVariant instead.

Also didn't want to make an overload as that would likely introduce some weirdness where passing the wrong fallback value would call the wrong version.

**Comment by Paxxi** [comment]
I ran it through clang-format to fix the formatting so there's plenty of changes now. Stuck them in a separate commit.

**Pull request merged**